src/main/java/io/vertx/core/http/impl/Http2HeadersAdaptor.java

```diff
@@ -113,29 +98,27 @@ public boolean isEmpty() {
 
   @Override
   public Set<String> names() {
-    if (names == null) {
-      names = new AbstractSet<String>() {
-        @Override
-        public Iterator<String> iterator() {
-          Iterator<CharSequence> it = headers.names().iterator();
-          return new Iterator<String>() {
-            @Override
-            public boolean hasNext() {
-              return it.hasNext();
-            }
-            @Override
-            public String next() {
-              return it.next().toString();
-            }
-          };
-        }
-        @Override
-        public int size() {
-          return headers.size();
-        }
-      };
-    }
-    return names;
+    Set<CharSequence> nameSet = headers.names();
+    return new AbstractSet<String>() {
+      @Override
+      public Iterator<String> iterator() {
+        Iterator<CharSequence> it = nameSet.iterator();
+        return new Iterator<String>() {
+          @Override
+          public boolean hasNext() {
+            return it.hasNext();
+          }
+          @Override
+          public String next() {
+            return it.next().toString();
+          }
+        };
+      }
+      @Override
+      public int size() {
+        return nameSet.size();
```

Comment:
need to change this to ensure that the size() returns the correct value as you can see from testEntries() test